src/vmm/src/device_manager/persist.rs

```diff
@@ -360,16 +364,16 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                                   slot: &MMIODeviceInfo,
                                   event_manager: &mut EventManager|
          -> Result<(), Self::Error> {
-            dev_manager
-                .slot_sanity_check(slot)
-                .map_err(Error::DeviceManager)?;
-
             let restore_args = MmioTransportConstructorArgs {
                 mem: mem.clone(),
                 device,
             };
             let mmio_transport =
                 MmioTransport::restore(restore_args, state).map_err(|()| Error::MmioTransport)?;
+            dev_manager
+                .legacy_irq_allocator
+                .allocate_id()
+                .map_err(|e| Error::DeviceManager(super::mmio::Error::AllocatorError(e)))?;
```

Comment:
this allocates a new irq right? but how do we make sure that we get the same irq we got before snapshotting?
AFAICT, this is currently only used to simulate that the allocator has the same state as before snapshotting.

But I'm afraid this logic is a bit flaky. If the logic for irq allocation changes in vm-allocator, we'll break in Firecracker.

It's also that this is currently not needed, right?

Reply:
> It's also that this is currently not needed, right?
Exactly it is not needed right now.

>this allocates a new irq right? but how do we make sure that we get the same irq we got before snapshotting?
AFAICT, this is currently only used to simulate that the allocator has the same state as before snapshotting.

We do not, because we do not care, we are using the irq from the saved slot anyway. 

>But I'm afraid this logic is a bit flaky. If the logic for irq allocation changes in vm-allocator, we'll break in  Firecracker.

The current implementation of restore will not be impacted by the changes in allocation policy in vm-allocator, this will only matter if hotplug support will be added, and by then vm-allocator might support something similar to `AllocPoilicy::ExactMatch` for IDs as well.

Reply:
> We do not, because we do not care, we are using the irq from the saved slot anyway.

Then why do this extra work now?

> The current implementation of restore will not be impacted by the changes in allocation policy in vm-allocator, this will only matter if hotplug support will be added, and by then vm-allocator might support something similar to AllocPoilicy::ExactMatch for IDs as well.

Agreed. Until then, I suggest we remove the re-allocation post-restore. It's not needed at the moment and if we add hotplug afterwards, we may start to rely on this brittle allocation logic (without adding something like `AllocPoilicy::ExactMatch`) for ids as well.

I think it's best to not do this extra work now